examples/ffmpeg_info.cpp

```diff
@@ -0,0 +1,87 @@
+// The contents of this file are in the public domain. See LICENSE_FOR_EXAMPLE_PROGRAMS.txt
+/*
+
+    This is an example illustrating some of the functional ffmpeg wrapper APIs.
+    It demonstrates how to list the supported codecs, muxers, demuxers and protocols
+    in your installation of ffmpeg (or the one dlib built against).
+
+    It also demonstrates how to check ffmpeg library versions programmatically.
+*/
+
+#include <cstdio>
+#include <dlib/media/ffmpeg_utils.h>
+
+using namespace std;
+using namespace dlib;
+
+int main()
+{
+    // List all codecs supported by this installation of ffmpeg libraries
+    const auto codecs = ffmpeg::list_codecs();
+    printf("Supported codecs:\n");
+    for (const auto& codec : codecs)
+        printf("name : %-16s : encoding supported %i decoding supported %i\n", codec.codec_name.c_str(), codec.supports_encoding, codec.supports_decoding);
```

Comment:
Use `cout` in the example instead of `printf`.  printf is so type unsafe.  Like this compiles
```
    std::string str = "asdf";
    printf("%s\n", str);
```
but just gives nonsense. I know some compilers warn about it but still.  Not all.  People should use `std::format` for libfmt.  It's just way better.

Anyway, use `cout` in the examples.